docs/tutorials/demo_data_science_project.py

```diff
@@ -109,15 +111,28 @@
 ## Data Ingestion
 
 Now it's time to get some data. We will use the [Open-Meteo API](https://open-meteo.com/), where we can pull weather data from an API for free (as long as we are non-commercial) and without an API token.
+In order to prevent hitting the rate limits when repeatedly querying the API (and out of courtesy towards the operators of the API), the `_maybe_urlretrieve` function provides a simple local cache for the downloaded data.
 
 For training our toy model, we download the full weather data of Munich for the year 2010:
 """
 
+
 # %%
-outfile, _ = urllib.request.urlretrieve(
-    "https://archive-api.open-meteo.com/v1/archive?latitude=52.52&longitude=13.41&start_date=2010-01-01&end_date=2010-12-31&hourly=temperature_2m,relativehumidity_2m,rain,pressure_msl,surface_pressure,cloudcover,cloudcover_low,cloudcover_mid,cloudcover_high,windspeed_10m,windspeed_100m,winddirection_10m,winddirection_100m"
-)
+def _maybe_urlretrieve(url: str, filename: str) -> str:
+    # Avoid API rate limit errors by downloading to a fixed local location
+    destination = Path(tempfile.gettempdir()) / "lakefs-spec-tutorials" / filename
+    destination.parent.mkdir(exist_ok=True, parents=True)
```

Comment:
What's the lifecycle of that `tempdir`? Is there any manual cleanup necessary?

Reply:
Up to the host OS. In fact, cleanup is the opposite of what we want, in order to prevent the API from being accessed multiple times when rebuilding the notebook (which happens in different processes).

I know, it's kinda hacky, but if you run of diskspace on your `$TMPDIR`, you should blame your OS for not cleaning it up on reboot/schedule 😅

Reply:
But this gives the same directory on multiple interpreter sessions, i.e. it caches on a rebuild? (I don't quite understand how, which is why I'm asking)